presto-docs/src/main/sphinx/develop/client-request-filter.rst

```diff
@@ -0,0 +1,23 @@
+
+======================
+Client Request Filter
+======================
+
+Presto facilitates seamless integration of additional headers into client requests using the 
+Client Request Filter plugin via the AuthenticationFilter class, enabling efficient and dynamic header management.
+
+Implementation
+--------------
+
+The ``ClientRequestFilterFactory`` is responsible for creating instances of ``ClientRequestFilter``. It also defines 
+the name of the filter, which is subsequently used as the header name in client requests.
+
+``ClientRequestFilter``  interface provides two methods: ``getExtraHeaders()``,
+which returns header values as key-value pairs in a map and ``getHeaderNames()`` which returns a list of header names.
+
+The implementation of ``ClientRequestFilterFactory`` must be wrapped
+as a plugin and installed on the Presto cluster.
+
+After installing a plugin that implements ``ClientRequestFilterFactory`` on the coordinator, the ``AuthenticationFilter`` class passes the ``principal`` object to the request filter, which returns the header values as a map.
+
+Presto uses the name of the request filter to determine whether the header is present in the blocklist.
```

Comment:
What blocklist?  Mention the headers which are not allowed to be overidden.  Link to the Java source of headers that may be overidden.

Reply:
Added.